### ISSUE TITLE
RIDE fails to load (traceback generated) if a plugin fails

### ISSUE THREAD

**Issue by memoselyk** [robotframework/RIDE#1598]
With RIDE 1.5.

I came across this issue, while trying to open two RIDE instances.
The `LogPlugin` failed to load, due to a shared file being _used by other process_, and it caused RIDE to fail, with the following traceback

> Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\robotide\__init__.py", line 73, in main
    _run(inpath, not noupdatecheck, debug_console)
  File "C:\Python27\lib\site-packages\robotide\__init__.py", line 99, in _run
    ride = RIDE(inpath, updatecheck)
  File "C:\Python27\lib\site-packages\robotide\application\application.py", line 42, in __init__
    wx.App.__init__(self, redirect=False)
  File "C:\Python27\lib\site-packages\wx-2.8-msw-unicode\wx\_core.py", line 7981, in __init__
    self._BootstrapApp()
  File "C:\Python27\lib\site-packages\wx-2.8-msw-unicode\wx\_core.py", line 7555, in _BootstrapApp
    return _core_.PyApp__BootstrapApp(*args, **kwargs)
  File "C:\Python27\lib\site-packages\robotide\application\application.py", line 55, in OnInit
    coreplugins.get_core_plugins())
  File "C:\Python27\lib\site-packages\robotide\application\pluginloader.py", line 30, in __init__
    standard_classes + self._find_classes(load_dirs) ]
  File "C:\Python27\lib\site-packages\robotide\application\pluginconnector.py", line 23, in PluginFactory
    return BrokenPlugin(utils.get_error_details(), plugin_class)
  File "C:\Python27\lib\site-packages\robotide\application\pluginconnector.py", line 69, in __init__
    + 'Error: ' + error
TypeError: cannot concatenate 'str' and 'tuple' objects

The problem is caused because `robot`'s `get_error_details` method returns a tuple with exception message and the produced traceback, since RF 2.9.

**Comment by memoselyk** [comment]
The triggering failure of this problem, makes me wonder if that should be more gracefully handled.

> WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users
28d5ddd-48e3-42db-b97f-e1172c48d52f-ride.log'
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\robotide\application\pluginconnector.py", line 21, in PluginFactory
    plugin = plugin_class(application)
  File "C:\Python27\lib\site-packages\robotide\log\log.py", line 44, in __init__
    self._remove_old_log_files()
  File "C:\Python27\lib\site-packages\robotide\log\log.py", line 56, in _remove_old_log_files
    os.remove(fname)

**Comment by HelioGuilherme66** [comment]
I cannot reproduce this with RF 3.0 and RIDE 1.5.2.1 in Linux.

**Comment by memoselyk** [comment]
You won't be able to reproduce the error by opening two instances of RIDE in other OS than Windows. But you could if you have an _artificially_ broken plugin. 

I just was looking at file `src/robotide/log/log.py`, the underlying issue is that `WindowsError`, which not an `IOError`, is being raised. 

BTW... I don't think that the installed RF should relevant since used `get_error_details` method is the one from the bundled RF.

**Comment by HelioGuilherme66** [comment]
I actually think I did once detected and fixed that exception.
You could just update the PR by adding the exception like in:
```
except OSError or IOError or WindowsError as e:
```
or in this format:
```
except (OSError, IOError, WindowsError) as e:
```

**Comment by memoselyk** [comment]
My bad, the `WindowsError` was fixed in 1.5.1. But there is still the other issue address by PR #1599.

**Comment by yanne** [comment]
Fixed by merging #1599

**Comment by jypandey1990** [comment]
Yaa, adding 
except (OSError, IOError, WindowsError) as e:
worked.
I had trouble while opening two instance of RIDE.
